### PR TITLE
Add DataFrame aggregations

### PR DESCRIPTION
**Pull request by sinhrks**
``DataFrame`` now supports same aggfuncs as ``Series``. ``DataFrame.std`` is ugly, should be revisited after #482.

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -696,6 +696,99 @@ def to_csv(self, filename, **kwargs):
...
+            raise ValueError('No axis named {0}'.format(axis))
+        # convert to numeric axis
+        return {None: 0, 'index': 0, 'columns': 1}.get(axis, axis)
+
+    @wraps(pd.DataFrame.sum)
+    def sum(self, axis=None):
+        axis = self._validate_axis(axis)
+        if axis == 1:
+            f = lambda x: x.sum(axis=1)
+            return map_partitions(f, self.columns, self)
```
Comment (mrocklin) [inline dask/dataframe/core.py]: Is the `columns` information correct here?

File: dask/dataframe/core.py [hunk 2]
```diff
@@ -696,6 +696,99 @@ def to_csv(self, filename, **kwargs):
...
+        else:
+            chunk = lambda df: df.sum()
+            agg = lambda df: df.groupby(level=0).sum()
+            return aca([self], chunk=chunk, aggregate=agg, columns=self.columns)
```
Comment (mrocklin) [inline dask/dataframe/core.py]: Same here actually, it seems like we want the result to be a `Series` with `name=None`

Comment (sinhrks) [inline dask/dataframe/core.py]: Yes, it should return ``Series``. Revisit after #512.

**Comment by sinhrks** [comment]
Updated based on #512. Now ready for review.

File: dask/dataframe/core.py [hunk 3]
```diff
@@ -747,6 +747,100 @@ def to_csv(self, filename, **kwargs):
...
+            raise ValueError('No axis named {0}'.format(axis))
+        # convert to numeric axis
+        return {None: 0, 'index': 0, 'columns': 1}.get(axis, axis)
+
+    @wraps(pd.DataFrame.sum)
+    def sum(self, axis=None):
+        axis = self._validate_axis(axis)
+        if axis == 1:
+            f = lambda x: x.sum(axis=1)
+            return self.map_partitions(f, self.columns, return_type=Series)
```
Comment (mrocklin) [inline dask/dataframe/core.py]: Hrm, the `columns=` kwarg here is troublesome.  The result should be a `Series` with `name=None`.  Currently `map_partitions` is a bit broken in that it only accepts a `columns=` kwarg and defaults to `self.columns` if that value is None.

File: dask/dataframe/core.py [hunk 4]
```diff
@@ -747,6 +747,100 @@ def to_csv(self, filename, **kwargs):
...
 
+    @wraps(pd.DataFrame._get_numeric_data)
+    def _get_numeric_data(self, how='any', subset=None):
+        return map_partitions(pd.DataFrame._get_numeric_data, self.columns, self)
+
+    def _validate_axis(self, axis=0):
+        if axis not in (0, 1, 'index', 'columns', None):
+            raise ValueError('No axis named {0}'.format(axis))
+        # convert to numeric axis
+        return {None: 0, 'index': 0, 'columns': 1}.get(axis, axis)
```
Comment (mrocklin) [inline dask/dataframe/core.py]: This looks like a good idea.

**Comment by mrocklin** [comment]
Other than the `map_partitions(..., columns=...)` issue this looks good to me.  The `map_partitions` issue is separate and pre-existing so I could also handle it separately.

**Comment by sinhrks** [comment]
Thanks. I'll wait for your ``map_partitions`` fix.

BTW, it's better to test ``DataFrame.columns`` and ``Series.name`` are correctly handled. Prepared #536.

**Comment by sinhrks** [comment]
This also ready. Removed ``return_type`` from `aca``.

**Comment by mrocklin** [comment]
Also looks good.  Merging.

**Pull request merged**